gremlin-go/driver/bytecode.go

```diff
@@ -122,16 +126,24 @@ func (bytecode *bytecode) convertArgument(arg interface{}) (interface{}, error)
 		return newSlice, nil
 	default:
 		switch v := arg.(type) {
-		case binding:
-			convertedValue, err := bytecode.convertArgument(v.value)
+		case *Binding:
+			convertedValue, err := bytecode.convertArgument(v.Value)
 			if err != nil {
 				return nil, err
 			}
-			bytecode.bindings[v.key] = v.value
-			return binding{
-				key:   v.key,
-				value: convertedValue,
+			bytecode.bindings[v.Key] = v.Value
+			return &Binding{
+				Key:   v.Key,
+				Value: convertedValue,
 			}, nil
```

Comment:
```suggestion
			return newBinding(v.Key, convertedValue), nil
```

Reply:
Ah, I'm actually not sure if I really needed the `newBinding` constructor, since we can create Bindings via structs, it if left from an attempt to set up the `Bindings.Of` call differently. Perhaps this can be discussed.